devicetypes/keen-home/keen-home-smart-vent.src/keen-home-smart-vent.groovy

```diff
@@ -477,47 +185,25 @@ def configure() {
-        // bind reporting clusters to hub
-            //commenting out switch cluster bind as using wrapper onOffConfig of zigbee class
-        //"zdo bind 0x${device.deviceNetworkId} 1 1 0x0006 {${device.zigbeeId}} {}", "delay 500",
-        "zdo bind 0x${device.deviceNetworkId} 1 1 0x0008 {${device.zigbeeId}} {}", "delay 500",
-        "zdo bind 0x${device.deviceNetworkId} 1 1 0x0402 {${device.zigbeeId}} {}", "delay 500",
-        "zdo bind 0x${device.deviceNetworkId} 1 1 0x0403 {${device.zigbeeId}} {}", "delay 500",
-        "zdo bind 0x${device.deviceNetworkId} 1 1 0x0001 {${device.zigbeeId}} {}", "delay 500"
-
-        // configure report commands
-        // zcl global send-me-a-report [cluster] [attr] [type] [min-interval] [max-interval] [min-change]
-
-        // report with these parameters is preconfigured in firmware, can be overridden here
-        // vent on/off state - type: boolean, change: 1
-        // "zcl global send-me-a-report 6 0 0x10 5 60 {01}", "delay 200",
-        // "send 0x${device.deviceNetworkId} 1 1", "delay 1500",
-
-        // report with these parameters is preconfigured in firmware, can be overridden here
-        // vent level - type: int8u, change: 1
-        // "zcl global send-me-a-report 8 0 0x20 5 60 {01}", "delay 200",
-        // "send 0x${device.deviceNetworkId} 1 1", "delay 1500",
+    def cmds = [
+            zigbee.temperatureConfig(30, 300) +
+                    zigbee.addBinding(zigbee.ONOFF_CLUSTER) +
+                    zigbee.addBinding(zigbee.LEVEL_CONTROL_CLUSTER) +
+                    zigbee.configureReporting(PRESSURE_MEASUREMENT_CLUSTER, 0x0020, DataType.UINT32, 600, 21600, 0x010000, [mfgCode: MFG_CODE]) +
+                    zigbee.configureReporting(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0021, DataType.UINT8, 600, 21600, 0x01) // battery precentage
+    ]
 
-        // report with these parameters is preconfigured in firmware, can be overridden here
-        // temperature - type: int16s, change: 0xA = 10 = 0.1C
-        // "zcl global send-me-a-report 0x0402 0 0x29 60 60 {0A00}", "delay 200",
-        // "send 0x${device.deviceNetworkId} 1 1", "delay 1500",
+    return delayBetween(cmds) + refresh()
+}
 
-        // report with these parameters is preconfigured in firmware, can be overridden here
-        // keen home custom pressure (tenths of Pascals) - type: int32u, change: 1 = 0.1Pa
-        // "zcl mfg-code 0x115B", "delay 200",
-        // "zcl global send-me-a-report 0x0403 0x20 0x22 60 60 {010000}", "delay 200",
-        // "send 0x${device.deviceNetworkId} 1 1", "delay 1500",
 
-        // report with these parameters is preconfigured in firmware, can be overridden here
-        // battery - type: int8u, change: 1
-        // "zcl global send-me-a-report 1 0x21 0x20 60 3600 {01}", "delay 200",
-        // "send 0x${device.deviceNetworkId} 1 1", "delay 1500",
-    ]
+private boolean isObstructed() {
+    def currentState = device.currentValue("switch")
```

Comment:
Where does this get set to `obstructed`?

Reply:
https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/45127/files#diff-bea535a5e9f948f1d19a1a80bb0be167L215

Reply:
Ah so this is just a workaround for existing values that are out there, it won't happen anymore?

Reply:
Right.